File: scripts/_strip_hardcoded_ratings.py
"""One-off: replace common hardcoded Rating stat literals in customer-web."""
import re
from pathlib import Path

root = Path(__file__).resolve().parents[1] / "apps" / "customer-web"
dash = "\u2014"
patterns = [
    (re.compile(r"\{ value: '\*4\.\d', label: 'Rating' \}"), f"{{ value: '{dash}', label: 'Rating' }}"),
    (re.compile(r'\{ value: "\*4\.\d", label: "Rating" \}'), f'{{ value: "{dash}", label: "Rating" }}'),
    (re.compile(r"\{ value: '4\.\d', label: 'Rating', icon:"), f"{{ value: '{dash}', label: 'Rating', icon:"),
    (re.compile(r'\{ value: "4\.\d", label: "Rating", icon:'), f'{{ value: "{dash}", label: "Rating", icon:'),
    (re.compile(r"\{ value: '4\.\d', label: 'Stays', icon:"), f"{{ value: '{dash}', label: 'Stays', icon:"),
    (re.compile(r"\{ value: '4\.\d', label: 'Café', icon:"), f"{{ value: '{dash}', label: 'Café', icon:"),
]
count = 0
for p in root.rglob("*.tsx"):
    t = p.read_text(encoding="utf-8")
    nt = t
    for rx, rep in patterns:
        nt = rx.sub(rep, nt)
    if nt != t:
        p.write_text(nt, encoding="utf-8")
        count += 1
        print(p.relative_to(root))
print("files updated:", count)

File: scripts/_strip_hardcoded_ratings.mjs
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const root = path.join(__dirname, '..', 'apps', 'customer-web');
const dash = '\u2014';

const patterns = [
  [/\{ value: '\*4\.\d', label: 'Rating' \}/g, `{ value: '${dash}', label: 'Rating' }`],
  [/\{ value: '\*4\.\d', label: "Rating" \}/g, `{ value: '${dash}', label: "Rating" }`],
  [/\{ value: '4\.\d', label: 'Rating', icon:/g, `{ value: '${dash}', label: 'Rating', icon:`],
  [/\{ value: "4\.\d", label: "Rating", icon:/g, `{ value: "${dash}", label: "Rating", icon:`],
  [/\{ value: '4\.\d', label: 'Stays', icon:/g, `{ value: '${dash}', label: 'Stays', icon:`],
  [/\{ value: '4\.\d', label: 'Café', icon:/g, `{ value: '${dash}', label: 'Café', icon:`],
];

function walk(dir, acc = []) {
  for (const e of fs.readdirSync(dir, { withFileTypes: true })) {
    const p = path.join(dir, e.name);
    if (e.isDirectory()) walk(p, acc);
    else if (p.endsWith('.tsx')) acc.push(p);
  }
  return acc;
}

let n = 0;
for (const p of walk(root)) {
  let t = fs.readFileSync(p, 'utf8');
  let nt = t;
  for (const [rx, rep] of patterns) nt = nt.replace(rx, rep);
  if (nt !== t) {
    fs.writeFileSync(p, nt, 'utf8');
    console.log(path.relative(root, p));
    n++;
  }
}
console.log('files updated:', n);
